src/Services/DepartmentValidatorService.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+namespace App\Services;
+
+use App\Entity\Department;
+use Symfony\Component\Validator\Validator\ValidatorInterface;
+
+class DepartmentValidatorService
+{
+    private ValidatorInterface $validator;
+
+    public function __construct(ValidatorInterface $validator)
+    {
+        $this->validator = $validator;
+    }
+
+    public function validateDepartmentData(array $data): array
+    {
+        $department = new Department();
+        $department->setTitle($data['title']);
+        $department->setSlug($data['slug']);
+        $department->setDescription($data['description']);
+        $department->setStatus(1);
+        $errors = $this->validator->validate($department);
+
+        if (count($errors)) {
+            $errorMessages = [];
+            foreach ($errors as $error) {
+                $errorMessages[] = $error->getMessage();
+            }
+
+            return [
+                'status' => false,
+                'message' => $errorMessages,
+            ];
+        }
+
+        return [
+            'status' => true,
+        ];
```

Comment:
We are duplicating this logic in all validator services. May be we can define an abstract/parent class & can add this method there.